client/packages/host/src/utils.ts

```diff
@@ -1,7 +1,24 @@
-import { ReactNode } from 'react';
+import { ReactNode, useState, useEffect } from 'react';
 
 export const getBadgeProps = (value?: number) => ({
   badgeContent: (value ?? 0) as ReactNode,
   max: 99,
   color: 'primary' as 'primary' | 'default',
 });
+
+export const useDeviceDetect = () => {
```

Comment:
rather than a new hook, just use `Capacitor.isNativePlatform()`
I'm nervous about the userAgent matching.. even though it would be totally fine

Reply:
Ok - good point @CarlosNZ the `isNativePlatform()` won't detect as intended if you are running a browser on a tablet.
In that case, could you please use

```
    if(EnvUtils.platform === Platform.Android) return;
```

I've tested this on a browser on android and it works as expected.
Perhaps also build out the user agent check in EnvUtils and add in `iOS` too?

Reply:
> rather than a new hook, just use `Capacitor.isNativePlatform()` I'm nervous about the userAgent matching.. even though it would be totally fine

Yeah, ideally it would work for any touch screen environment, regardless of OS/browser/build. Perhaps we should add a specific `isTouchScreen` property to `EnvUtils`?

Reply:
true - that's what we're looking to detect here. I liked the use of 'ontouchstart' that you linked to.
This seems to work..

```
const isTouchScreen = 'ontouchstart' in document.documentElement;

export const EnvUtils = {
  // Using isProduction rather than isDevelopment, as we also use a testing
  // environment while running jest, so easier to check !isProduction, generally.
  isProduction: (): boolean => process.env['NODE_ENV'] === 'production',
  isTouchScreen,
  mapRoute,
  platform,
  printFormat: PrintFormat.Html, // platform === Platform.Android ? PrintFormat.Html : PrintFormat.Pdf,
};
```

```
    if (EnvUtils.isTouchScreen) return;
```